### DISCUSSION TITLE
How to extract a polygon from an image

### DISCUSSION THREAD

**Discussion by alexsmithbr** [ImageMagick/ImageMagick#5528] [Help]
At https://legacy.imagemagick.org/discourse-server/viewtopic.php?p=109604 an user asked how to do it and even came up with some results. I would just like to document here my approach to the case. He wanted to extract a subimage in a non-fixed form i.e. not squared.

```
convert input_image.png \( +clone -fill Black -colorize 100 -fill White -draw "polygon x1,y1 x2,y2 x3,y3 x4,y4" \) -compose Multiply -composite -trim output_image.png
```

We can break the command above to make it more readable:

```
convert input_image.png \
   \( \
      +clone -fill Black -colorize 100 \
      -fill White -draw "polygon x1,y1 x2,y2 x3,y3 x4,y4" \
   \) \
   -compose Multiply -composite \
   -trim \
   output_image.png
```

What happens here is:

1. input_image.png is the input image.
2. +clone will clone the contents of the image to memory.
3. -fill Black -colorize 100 will paint the cloned image entirely Black.
4. -fill White -draw "polygon ..." will draw a polygon following all specified x,y coordinates, filling its inner side in white in the end.
5. -compose Multiply -composite will apply a composition (i.e. operate over cloned and original images) using the Multiply approach, meaning pixels will have their colors multiplied.
6. -trim will trim the final image to its edges.
7. output_image.png will store the resulting image.

Hope it helps someone out there.

**Comment by snibgo** [comment]
Thanks for sharing.

If we replace "`-compose Multiply -composite`" with "`-alpha off -compose CopyOpacity -composite`" then all the pixels outside the polygon will be made transparent.
